### ISSUE TITLE
API: bot.findBlock fallback?

### ISSUE THREAD

**Issue by Pietro210** [andrewrk/mineflayer#297]
This is not really an issue but I noticed that a lot of examples use a simple function to find blocks around the player and I think it's a common requirement for bots, however Mineflayer has no built-in method for that.

[Darthfett/mineflayer-blockFinder](https://github.com/Darthfett/mineflayer-blockFinder) basically does the same thing and it's probably more flexible and efficient, but wouldn't it be better if there was a fallback method?

It would be really simple to implement and it would be used the same way, so it can be easily overridden by the blockFinder plugin without changing any code if the user wants a more efficient version of the same method, this would be specified in the documentation.

This would also simplify some of the examples by removing some duplicate code.

I've already made a method that behaves like that for one of my bots, so if it is a good idea I could submit a pull request.

**Comment by roblabla** [comment]
Mineflayer should be the lowest API necessary to implement higher-level stuff. A block finder is fairly high-level and can be implemented without too much trouble with the methods mineflayer exposes. As such, it shouldn't be added in. If a user needs a blockFinder, he should use a mineflayer plugin to do it (or re-invent the wheel, which is what the examples do ^^).

**Comment by rom1504** [comment]
I agree with @roblabla but maybe we should have one example (with its own package.json) including all the official mineflayer plugins and showing how to use them.

**Comment by roblabla** [comment]
:+1: for having a complete example with tons of plugins. Make sure it has a thorough documentation so people understand the additional functions come from plugins :D

**Comment by Pietro210** [comment]
That's a good idea, just one more thing to consider though, are you sure there isn't a case when it would be useful to know the presence of a block nearby before/after performing certain actions?

I'm thinking of something like when you try to use a bed and there are mobs nearby and the client doesn't let you do that, of course mobs are entities and not blocks but you see what I mean..

In that case a reusable function would be useful, and we might as well expose it as part of the API

**Comment by rom1504** [comment]
So in the end I added @Pietro210's simple findBlock function to blocks.js.

I agree we should not put too many things in mineflayer, but adding this function makes examples not repeat code again and again and it's only a few lines of code in blocks.js. I added a notice in the doc ( https://github.com/andrewrk/mineflayer/blob/master/doc/api.md#botfindblockoptions ) to prefer mineflayer-blockfinder plugin for more complicated uses.
